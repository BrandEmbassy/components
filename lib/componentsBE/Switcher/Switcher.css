.Switcher {
  position: relative;
  display: flex;
  width: auto;
  height: 24px;
  line-height: 24px;
  cursor: pointer;

  & .Label {
    position: relative;
    flex: 0 1 auto;
    height: 24px;
    line-height: 26px;
    padding: 0 8px;
    font-size: 13px;
    transition: all .18s;

    & on {
      color: #999999;
    }

    & off {
      font-weight: 700;
      color: #999999;
    }
  }

  &__switch {
    position: relative;
    flex: 0 1 auto;
    width: 44px;
    height: 24px;
    box-shadow: 0 1px 2px rgba(0, 0, 0, 0.1);
    border: 1px solid #b3b3b3;
    border-radius: 4px;
    background-color: #ffffff;

    &:after {
      content: '';
      position: absolute;
      top: 0;
      right: 0;
      left: auto;
      display: block;
      width: 16px;
      height: 16px;
      margin: 3px;
      border-radius: 2px;
      background-color: #999;
      transition: all .18s;
    }
  }

  &--isActive {
    .Switcher {
      &__label {
        &--on {
          font-weight: 700;
          color: #36a909;
        }
        &--off {
          font-weight: 400;
        }
      }
      &__switch {
        &:after {
          right: 20px;
          background-color: #36a909;
        }
      }
    }
  }
}
