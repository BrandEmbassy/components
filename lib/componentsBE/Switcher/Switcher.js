'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _bind = require('classnames/bind');

var _bind2 = _interopRequireDefault(_bind);

var _Switcher = require('./Switcher.css');

var _Switcher2 = _interopRequireDefault(_Switcher);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

var cx = _bind2.default.bind(_Switcher2.default);

var Switcher = function (_Component) {
  _inherits(Switcher, _Component);

  function Switcher(props) {
    _classCallCheck(this, Switcher);

    var _this = _possibleConstructorReturn(this, (Switcher.__proto__ || Object.getPrototypeOf(Switcher)).call(this, props));

    _this.state = { isActive: _this.props.isActive };
    return _this;
  }

  _createClass(Switcher, [{
    key: 'render',
    value: function render() {
      var _this2 = this;

      var isActive = this.props.isActive;


      var className = cx(_Switcher2.default.Switcher, {
        isActive: isActive
      });

      return _react2.default.createElement(
        'div',
        {
          onClick: function onClick(e) {
            e.stopPropagation();
            _this2.setState({ isActive: !_this2.state.isActive });
          },
          className: className },
        _react2.default.createElement(
          'div',
          { className: 'Label Label--on' },
          'On'
        ),
        _react2.default.createElement('div', { className: 'Switch' }),
        _react2.default.createElement(
          'div',
          { className: 'Label Label--off' },
          'Off'
        )
      );
    }
  }]);

  return Switcher;
}(_react.Component);

exports.default = Switcher;