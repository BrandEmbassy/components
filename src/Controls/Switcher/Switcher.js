import React, { Component } from 'react';
import styles from './Switcher.css';

export default class Switcher extends Component {
  render() {
    return (
      <div className={styles.Switcher}>
        switcher
      </div>
    );
  }
}

