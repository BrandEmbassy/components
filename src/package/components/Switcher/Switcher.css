.Switcher {
  position: relative;
  display: flex;
  width: auto;
  height: 24px;
  line-height: 24px;
  cursor: pointer;
  margin: 4px;

  & .Label {
    position: relative;
    flex: 0 1 auto;
    height: 24px;
    line-height: 26px;
    padding: 0 8px;
    font-size: 13px;
    transition: all .18s;

    &.LabelOn {
      color: #999999;
    }

    &.Label--off {
      font-weight: 700;
      color: #999999;
    }
  }

  & .Switch {
    position: relative;
    flex: 0 1 auto;
    width: 44px;
    height: 24px;
    box-shadow: 0 1px 2px rgba(0, 0, 0, 0.1);
    border: 1px solid #b3b3b3;
    border-radius: 4px;
    background-color: #ffffff;

    &:after {
      content: '';
      position: absolute;
      top: 0;
      right: 0;
      left: auto;
      display: block;
      width: 16px;
      height: 16px;
      margin: 4px 3px;
      border-radius: 2px;
      background-color: #999;
      transition: all .18s;
    }
  }

  &.isActive {
    &.Switcher {
      & .Label {
        &.LabelOn {
          font-weight: 700;
          color: #36a909;
          padding-right: 7px;
        }
        &.LabelOff {
          font-weight: 400;
          color: #999999;
        }
      }
      & .Switch {
        &:after {
          right: 20px;
          background-color: #36a909;
        }
      }
    }
  }
}
