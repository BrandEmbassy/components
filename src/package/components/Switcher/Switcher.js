// @flow

import * as React from "react";
// @flow-skip-next-line
import classNames from "classnames/bind";
import styles from "./Switcher.module.css";

const cx = classNames.bind(styles);

type Props = {
  title: string,
  styleName: string,
  isActive: boolean,
  SwitcherClick: Function
};

export default class Switcher extends React.Component<Props> {
  render() {
    const { styleName, isActive, SwitcherClick } = this.props;
    const { Switcher, Switch, Label, LabelOn, LabelOff } = styles;

    const On = cx(Label, LabelOn);
    const Off = cx(Label, LabelOff);
    const className = cx(Switcher, styleName, {
      isActive
    });

    return (
      <div className={className} onClick={SwitcherClick} data-jest="click">
        <div className={On}>On</div>
        <div className={Switch} />
        <div className={Off}>Off</div>
      </div>
    );
  }
}
