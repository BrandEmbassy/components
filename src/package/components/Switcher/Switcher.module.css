.Switcher {
  position: relative;
  display: flex;
  width: auto;
  height: 24px;
  line-height: 24px;
  cursor: pointer;
  margin: 4px;
}
.Switcher .Label {
  position: relative;
  flex: 0 1 auto;
  height: 24px;
  line-height: 26px;
  padding: 0 8px;
  font-size: 13px;
  transition: all 0.18s;
}
.Switcher .Label.LabelOn {
  color: #999999;
}
.Switcher .Label.Label--off {
  font-weight: 700;
  color: #999999;
}
.Switcher .Switch {
  position: relative;
  flex: 0 1 auto;
  width: 44px;
  height: 24px;
  box-shadow: 0 1px 2px rgba(0, 0, 0, 0.1);
  border: 1px solid #b3b3b3;
  border-radius: 4px;
  background-color: #ffffff;
}
.Switcher .Switch:after {
  content: "";
  position: absolute;
  top: 0;
  right: 0;
  left: auto;
  display: block;
  width: 16px;
  height: 16px;
  margin: 4px 3px;
  border-radius: 2px;
  background-color: #999;
  transition: all 0.18s;
}
.Switcher.isActive.Switcher .Label.LabelOn {
  font-weight: 700;
  color: #36a909;
  padding-right: 7px;
}
.Switcher.isActive.Switcher .Label.LabelOff {
  font-weight: 400;
  color: #999999;
}
.Switcher.isActive.Switcher .Switch:after {
  right: 20px;
  background-color: #36a909;
}
