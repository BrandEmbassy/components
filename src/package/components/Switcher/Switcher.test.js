import React from 'react'
import Switcher from './Switcher'

describe('Switcher', function () {
  it('should render switcher ON', () => {
    const wrapper = shallow(<Switcher isActive />)
    expect(wrapper).toMatchSnapshot()
  })
  it('should render switcher OFF', () => {
    const wrapper = shallow(<Switcher />)
    expect(wrapper).toMatchSnapshot()
  })
  it('should call onClick', () => {
    const clickHandler = jest.fn()
    const wrapper = shallow(<Switcher SwitcherClick={clickHandler} />)
    wrapper.find('[data-jest="click"]').simulate('click')
    expect(clickHandler).toBeCalled()
  })
})
