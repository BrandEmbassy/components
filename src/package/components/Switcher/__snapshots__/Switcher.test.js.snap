// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Switcher should render switcher OFF 1`] = `
<div
  className=""
  data-jest="click"
>
  <div
    className=""
  >
    On
  </div>
  <div />
  <div
    className=""
  >
    Off
  </div>
</div>
`;

exports[`Switcher should render switcher ON 1`] = `
<div
  className="isActive"
  data-jest="click"
>
  <div
    className=""
  >
    On
  </div>
  <div />
  <div
    className=""
  >
    Off
  </div>
</div>
`;
